2017-04-08 00:58:59 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/Adam/git/SEPT_AppointmentBook/db/database.mv.db [1.4.194/7]" [50000-194]
	at org.h2.message.DbException.get(DbException.java:168)
	at org.h2.message.DbException.convert(DbException.java:295)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:95)
	at org.h2.mvstore.MVStore.panic(MVStore.java:378)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:361)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2930)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:155)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:100)
	at org.h2.engine.Database.getPageStore(Database.java:2474)
	at org.h2.engine.Database.open(Database.java:697)
	at org.h2.engine.Database.openDatabase(Database.java:276)
	at org.h2.engine.Database.<init>(Database.java:270)
	at org.h2.engine.Engine.openSession(Engine.java:64)
	at org.h2.engine.Engine.openSession(Engine.java:176)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:154)
	at org.h2.engine.Engine.createSession(Engine.java:137)
	at org.h2.engine.Engine.createSession(Engine.java:27)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:354)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:116)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:100)
	at org.h2.Driver.connect(Driver.java:69)
	at java.sql.DriverManager.getConnection(Unknown Source)
	at java.sql.DriverManager.getConnection(Unknown Source)
	at bookingSystem.Database.getDBConnection(Database.java:33)
	at bookingSystem.Database.execQuery(Database.java:100)
	at bookingSystem.Database.loadFromDB(Database.java:68)
	at bookingSystem.SystemDriver.loadSystem(SystemDriver.java:67)
	at bookingSystem.Main.main(Main.java:12)
Caused by: org.h2.jdbc.JdbcSQLException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/Adam/git/SEPT_AppointmentBook/db/database.mv.db [1.4.194/7]" [50000-194]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:345)
	... 28 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:C:/Users/Adam/git/SEPT_AppointmentBook/db/database.mv.db [1.4.194/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:765)
	at org.h2.mvstore.FileStore.open(FileStore.java:173)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:347)
	... 23 more
2017-04-08 00:59:37 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: The object is already closed [90007-194]
2017-04-08 01:00:04 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: The object is already closed [90007-194]
2017-04-08 01:00:26 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: The object is already closed [90007-194]
2017-04-08 01:01:04 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: The object is already closed [90007-194]
2017-04-08 01:01:16 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: The object is already closed [90007-194]
2017-04-08 02:12:16 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "NAME[*]"; SQL statement:
NAME [42000-194]
2017-04-08 02:22:09 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Data conversion error converting "'Hooch' (TIMESLOTS_COPY_3_0: EMPLOYEE INT)"; SQL statement:
 -- row #2 ('2017 03 26 13:00:00', 'Hooch', 'false') [22018-194]
2017-04-08 02:24:06 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Unknown data type: ; SQL statement:
ALTER TABLE TIMESLOTS
  ADD COLUMN EMPLOYEE_ID [50004-194]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:345)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.command.Parser.parseColumnWithType(Parser.java:4144)
	at org.h2.command.Parser.parseColumnForTable(Parser.java:3994)
	at org.h2.command.Parser.parseAlterTableAddColumn(Parser.java:5808)
	at org.h2.command.Parser.parseAlterTable(Parser.java:5528)
	at org.h2.command.Parser.parseAlter(Parser.java:4920)
	at org.h2.command.Parser.parsePrepared(Parser.java:346)
	at org.h2.command.Parser.parse(Parser.java:319)
	at org.h2.command.Parser.parse(Parser.java:291)
	at org.h2.command.Parser.prepareCommand(Parser.java:256)
	at org.h2.engine.Session.prepareLocal(Session.java:564)
	at org.h2.engine.Session.prepareCommand(Session.java:505)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1204)
	at org.h2.jdbc.JdbcStatement.executeInternal(JdbcStatement.java:170)
	at org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:158)
	at org.h2.server.web.WebApp.getResult(WebApp.java:1380)
	at org.h2.server.web.WebApp.query(WebApp.java:1053)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1015)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1002)
	at org.h2.server.web.WebThread.process(WebThread.java:164)
	at org.h2.server.web.WebThread.run(WebThread.java:89)
	at java.lang.Thread.run(Unknown Source)
2017-04-08 02:24:37 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "ALTER TABLE TIMESLOTS
  ADD COLUMN EMPLOYEE_ID INT NOT_NULL[*] "; SQL statement:
ALTER TABLE TIMESLOTS
  ADD COLUMN EMPLOYEE_ID int NOT_NULL [42000-194]
2017-04-08 02:27:43 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "ALTER TABLE TIMESLOTS
  ADD EMPLOYEE_ID INT FOREIGN[*] KEY REFERENCES EMPLOYEE(ID) "; SQL statement:
ALTER TABLE TIMESLOTS
  ADD EMPLOYEE_ID int FOREIGN KEY REFERENCES EMPLOYEE(ID) [42000-194]
2017-04-08 02:28:46 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "ALTER TABLE TIMESLOTS
  ALTER COLUMN EMPLOYEE_ID INT FOREIGN[*] KEY REFERENCES EMPLOYEE(ID) "; SQL statement:
ALTER TABLE TIMESLOTS
  ALTER COLUMN EMPLOYEE_ID int FOREIGN KEY REFERENCES employee(ID) [42000-194]
2017-04-08 02:32:01 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "CREATE TABLE TIMESLOTS (
    EMPLOYEE_ID INT NOT NULL FOREIGN[*] KEY REFERENCES EMPLOYEE(ID),
    DATE VARCHAR(255) NOT NULL,
    BOOKED BOOLEAN NOT NULL
) "; expected "AS, DEFAULT, GENERATED, NOT, NULL, AUTO_INCREMENT, BIGSERIAL, SERIAL, IDENTITY, NULL_TO_DEFAULT, SEQUENCE, SELECTIVITY, COMMENT, CONSTRAINT, PRIMARY, UNIQUE, NOT, NULL, CHECK, REFERENCES, ,, )"; SQL statement:
CREATE TABLE Timeslots (
    Employee_ID int NOT NULL FOREIGN KEY REFERENCES Employee(ID),
    Date varchar(255) NOT NULL,
    booked boolean not null
) [42001-194]
2017-04-08 02:35:18 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "EMPLOYEE_ID[*]"; expected "EXPLAIN, EXECUTE, {"; SQL statement:
EMPLOYEE_ID [42001-194]
2017-04-08 02:49:45 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "ALTER TABLE BOOKINGS
ADD EMPLOYEE_ID INT NOT NULL,[*]
FOREIGN KEY (EMPLOYEE_ID) REFERENCES EMPLOYEE(ID) "; SQL statement:
ALTER table BOOKINGS
add employee_id int not null,
FOREIGN KEY (employee_id) REFERENCES employee(ID) [42000-194]
2017-04-08 02:56:55 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Index "PRIMARY_KEY_27" belongs to constraint "CONSTRAINT_A"; SQL statement:
ALTER TABLE user
DROP PRIMARY KEY [90085-194]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:345)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.command.ddl.DropIndex.update(DropIndex.java:63)
	at org.h2.command.CommandContainer.update(CommandContainer.java:101)
	at org.h2.command.Command.executeUpdate(Command.java:258)
	at org.h2.jdbc.JdbcStatement.executeInternal(JdbcStatement.java:184)
	at org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:158)
	at org.h2.server.web.WebApp.getResult(WebApp.java:1380)
	at org.h2.server.web.WebApp.query(WebApp.java:1053)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1015)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1002)
	at org.h2.server.web.WebThread.process(WebThread.java:164)
	at org.h2.server.web.WebThread.run(WebThread.java:89)
	at java.lang.Thread.run(Unknown Source)
2017-04-08 02:58:01 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "DROP BOOKINGS[*] "; expected "TABLE, INDEX, USER, SEQUENCE, CONSTANT, TRIGGER, VIEW, ROLE, ALIAS, SCHEMA, ALL, DOMAIN, TYPE, DATATYPE, AGGREGATE"; SQL statement:
drop bookings [42001-194]
2017-04-08 02:59:06 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "ALTER TABLE USER
ADD COLUMN ID INT NOT NULL PRIMARY[*] KEY "; SQL statement:
alter table user
add column ID int NOT NULL primary key [42000-194]
2017-04-08 03:00:28 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "ALTER TABLE USER
ADD ID INT NOT NULL PRIMARY[*] KEY "; SQL statement:
alter table user
add ID int NOT NULL primary key [42000-194]
2017-04-08 03:02:14 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Column "ID" must not be nullable; SQL statement:
ALTER TABLE user
add PRIMARY KEY (ID) [90023-194]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:345)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.mvstore.db.MVTable.addIndex(MVTable.java:477)
	at org.h2.command.ddl.AlterTableAddConstraint.tryUpdate(AlterTableAddConstraint.java:147)
	at org.h2.command.ddl.AlterTableAddConstraint.update(AlterTableAddConstraint.java:77)
	at org.h2.command.CommandContainer.update(CommandContainer.java:101)
	at org.h2.command.Command.executeUpdate(Command.java:258)
	at org.h2.jdbc.JdbcStatement.executeInternal(JdbcStatement.java:184)
	at org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:158)
	at org.h2.server.web.WebApp.getResult(WebApp.java:1380)
	at org.h2.server.web.WebApp.query(WebApp.java:1053)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1015)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1002)
	at org.h2.server.web.WebThread.process(WebThread.java:164)
	at org.h2.server.web.WebThread.run(WebThread.java:89)
	at java.lang.Thread.run(Unknown Source)
2017-04-08 03:02:28 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Unknown data type: "NOT"; SQL statement:
ALTER TABLE user
alter column ID not null [50004-194]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:345)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.command.Parser.parseColumnWithType(Parser.java:4144)
	at org.h2.command.Parser.parseColumnForTable(Parser.java:3994)
	at org.h2.command.Parser.parseAlterTableAlterColumnType(Parser.java:5770)
	at org.h2.command.Parser.parseAlterTable(Parser.java:5739)
	at org.h2.command.Parser.parseAlter(Parser.java:4920)
	at org.h2.command.Parser.parsePrepared(Parser.java:346)
	at org.h2.command.Parser.parse(Parser.java:319)
	at org.h2.command.Parser.parse(Parser.java:291)
	at org.h2.command.Parser.prepareCommand(Parser.java:256)
	at org.h2.engine.Session.prepareLocal(Session.java:564)
	at org.h2.engine.Session.prepareCommand(Session.java:505)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1204)
	at org.h2.jdbc.JdbcStatement.executeInternal(JdbcStatement.java:170)
	at org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:158)
	at org.h2.server.web.WebApp.getResult(WebApp.java:1380)
	at org.h2.server.web.WebApp.query(WebApp.java:1053)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1015)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1002)
	at org.h2.server.web.WebThread.process(WebThread.java:164)
	at org.h2.server.web.WebThread.run(WebThread.java:89)
	at java.lang.Thread.run(Unknown Source)
2017-04-08 03:04:07 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Column "ID" contains null values; SQL statement:
ALTER TABLE user
alter column ID int not null [90081-194]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:345)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.command.ddl.AlterTableAlterColumn.checkNoNullValues(AlterTableAlterColumn.java:515)
	at org.h2.command.ddl.AlterTableAlterColumn.update(AlterTableAlterColumn.java:155)
	at org.h2.command.CommandContainer.update(CommandContainer.java:101)
	at org.h2.command.Command.executeUpdate(Command.java:258)
	at org.h2.jdbc.JdbcStatement.executeInternal(JdbcStatement.java:184)
	at org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:158)
	at org.h2.server.web.WebApp.getResult(WebApp.java:1380)
	at org.h2.server.web.WebApp.query(WebApp.java:1053)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1015)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1002)
	at org.h2.server.web.WebThread.process(WebThread.java:164)
	at org.h2.server.web.WebThread.run(WebThread.java:89)
	at java.lang.Thread.run(Unknown Source)
2017-04-08 03:06:17 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Column "JIMBOB" not found; SQL statement:
UPDATE
  user
SET
  id = '1'
WHERE
  username IS Jimbob [42122-194]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:345)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.expression.ExpressionColumn.optimize(ExpressionColumn.java:146)
	at org.h2.expression.Comparison.optimize(Comparison.java:180)
	at org.h2.command.dml.Update.prepare(Update.java:185)
	at org.h2.command.Parser.prepareCommand(Parser.java:261)
	at org.h2.engine.Session.prepareLocal(Session.java:564)
	at org.h2.engine.Session.prepareCommand(Session.java:505)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1204)
	at org.h2.jdbc.JdbcStatement.executeInternal(JdbcStatement.java:170)
	at org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:158)
	at org.h2.server.web.WebApp.getResult(WebApp.java:1380)
	at org.h2.server.web.WebApp.query(WebApp.java:1053)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1015)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1002)
	at org.h2.server.web.WebThread.process(WebThread.java:164)
	at org.h2.server.web.WebThread.run(WebThread.java:89)
	at java.lang.Thread.run(Unknown Source)
2017-04-08 03:24:59 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/Adam/git/SEPT_AppointmentBook/db/database.mv.db [1.4.194/7]" [50000-194]
	at org.h2.message.DbException.get(DbException.java:168)
	at org.h2.message.DbException.convert(DbException.java:295)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:95)
	at org.h2.mvstore.MVStore.panic(MVStore.java:378)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:361)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2930)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:155)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:100)
	at org.h2.engine.Database.getPageStore(Database.java:2474)
	at org.h2.engine.Database.open(Database.java:697)
	at org.h2.engine.Database.openDatabase(Database.java:276)
	at org.h2.engine.Database.<init>(Database.java:270)
	at org.h2.engine.Engine.openSession(Engine.java:64)
	at org.h2.engine.Engine.openSession(Engine.java:176)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:154)
	at org.h2.engine.Engine.createSession(Engine.java:137)
	at org.h2.engine.Engine.createSession(Engine.java:27)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:354)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:116)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:100)
	at org.h2.Driver.connect(Driver.java:69)
	at java.sql.DriverManager.getConnection(Unknown Source)
	at java.sql.DriverManager.getConnection(Unknown Source)
	at bookingSystem.Database.getDBConnection(Database.java:74)
	at bookingSystem.Database.getUsers(Database.java:106)
	at bookingSystem.Database.loadFromDB(Database.java:94)
	at bookingSystem.SystemDriver.loadSystem(SystemDriver.java:65)
	at bookingSystem.Main.main(Main.java:12)
Caused by: org.h2.jdbc.JdbcSQLException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/Adam/git/SEPT_AppointmentBook/db/database.mv.db [1.4.194/7]" [50000-194]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:345)
	... 28 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:C:/Users/Adam/git/SEPT_AppointmentBook/db/database.mv.db [1.4.194/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:765)
	at org.h2.mvstore.FileStore.open(FileStore.java:173)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:347)
	... 23 more
2017-04-08 03:56:54 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "ALTER TABLE BOOKINGS
ADD TIMESLOT INT NOT NULL
ADD[*] FOREIGN KEY (TIMESLOT) REFERENCES TIMESLOTS(ID) "; SQL statement:
alter table bookings
add timeslot int not null
ADD FOREIGN KEY (timeslot) REFERENCES timeslots(id) [42000-194]
2017-04-08 03:57:13 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "ALTER TABLE BOOKINGS
ADD TIMESLOT INT NOT NULL,[*]
ADD FOREIGN KEY (TIMESLOT) REFERENCES TIMESLOTS(ID) "; SQL statement:
alter table bookings
add timeslot int not null,
ADD FOREIGN KEY (timeslot) REFERENCES timeslots(id) [42000-194]
2017-04-08 03:57:27 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "ALTER TABLE BOOKINGS
ADD TIMESLOT INT NOT NULL,[*]
ADD FOREIGN KEY (TIMESLOT) REFERENCES TIMESLOTS(ID) "; SQL statement:
alter table bookings
add timeslot int not null,
ADD FOREIGN KEY (timeslot) REFERENCES timeslots(id) [42000-194]
2017-04-08 04:33:36 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Cannot parse "DATE" constant "07/07/1907" [22007-194]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:345)
	at org.h2.message.DbException.get(DbException.java:168)
	at org.h2.value.ValueDate.parse(ValueDate.java:81)
	at org.h2.value.Value.convertTo(Value.java:929)
	at org.h2.value.Value.getDate(Value.java:409)
	at org.h2.jdbc.JdbcResultSet.getDate(JdbcResultSet.java:452)
	at bookingSystem.Database.getUsers(Database.java:124)
	at bookingSystem.Database.loadFromDB(Database.java:96)
	at bookingSystem.SystemDriver.loadSystem(SystemDriver.java:66)
	at bookingSystem.Main.main(Main.java:12)
Caused by: java.lang.IllegalArgumentException: 07/07/1907
	at org.h2.util.DateTimeUtils.parseDateValue(DateTimeUtils.java:313)
	at org.h2.value.ValueDate.parse(ValueDate.java:79)
	... 7 more
